### PR TITLE
added GenericDao and GenericService

### PR DESCRIPTION
**Pull request by VitaliiLeshchenko**

### CODE REVIEW

File: src/main/webapp/WEB-INF/jsp/index.jsp [hunk 1]
```diff
@@ -1,16 +1,17 @@
 <%--
   Created by IntelliJ IDEA.
   User: VitaliiLeshchenko
-  Date: 24-Apr-20
-  Time: 5:53 PM
+  Date: 25-Apr-20
+  Time: 3:53 PM
   To change this template use File | Settings | File Templates.
 --%>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/jsp/index.jsp]: let's remove comments created by idea

File: src/main/webapp/WEB-INF/web.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        <servlet-name>index</servlet-name>
+        <servlet-class>mate.academy.internetshop.controllers.IndexController</servlet-class>
+    </servlet>
+    
+    <servlet-mapping>
+        <servlet-name>index</servlet-name>
+        <url-pattern>/index</url-pattern>
+    </servlet-mapping>
+
+</web-app>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/web.xml]: add empty line to the end of file

**Review by dmytro-hryhoruk — COMMENTED** [review]
Let's set up Travis CI so it will check your PR https://www.youtube.com/watch?v=BHSsWTbuow4

File: src/main/webapp/WEB-INF/jsp/index.jsp (hunk 1 above)
Comment (VitaliiLeshchenko) [reply src/main/webapp/WEB-INF/jsp/index.jsp]: done

File: src/main/webapp/WEB-INF/web.xml (hunk 2 above)
Comment (VitaliiLeshchenko) [reply src/main/webapp/WEB-INF/web.xml]: done

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
+.out/*
 target/*
```
Comment (dmytro-hryhoruk) [inline .gitignore:4]: same

File: pom.xml [hunk 4]
```diff
@@ -65,5 +79,4 @@
             </plugins>
         </pluginManagement>
     </build>
-
 </project>
```
Comment (dmytro-hryhoruk) [inline pom.xml]: and here

**Review by dmytro-hryhoruk — COMMENTED** [review]
https://www.youtube.com/watch?v=BHSsWTbuow4

File: .gitignore (hunk 3 above)
Comment (VitaliiLeshchenko) [reply .gitignore:4]: done

File: pom.xml (hunk 4 above)
Comment (VitaliiLeshchenko) [reply pom.xml]: done

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+language: java
+jdk:
+  - openjdk11
+script:
+  - mvn clean install
+cache:
+  directories:
+    - $HOME/.m2
```
Comment (dmytro-hryhoruk) [inline .travis.yml:8]: empty line)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**